code/modules/reagents/reagent_containers/food/cans.dm

```diff
@@ -24,6 +29,181 @@
 		return TRUE
 	. = ..()
 
+/obj/item/reagent_containers/food/drinks/cans/update_icon()
+	cut_overlays()
+	var/image/fuseoverlay = image('icons/obj/fuses.dmi', icon_state = "fuse_short")
+	switch(fuselength)
+		if(1 to 5)
+			if("x" in can_size_overrides)
+				fuseoverlay.pixel_x = can_size_overrides["x"]
+			if("y" in can_size_overrides)
+				fuseoverlay.pixel_y = can_size_overrides["y"]
+			add_overlay(fuseoverlay)
+		if(6 to INFINITY)
+			fuseoverlay.icon_state = "fuse_long"
+			if("x" in can_size_overrides)
+				fuseoverlay.pixel_x = can_size_overrides["x"]
+			if("y" in can_size_overrides)
+				fuseoverlay.pixel_y = can_size_overrides["y"]
```

Comment:
This is repeated code; we don't need to do it twice, just apply it after the switch.